cf-check/dump.c

```diff
@@ -154,23 +165,27 @@ static void print_struct_averages(
     else
     {
         // TODO: clean up Averages
+        const char *const *obnames = NULL;
         const Averages *const averages = value.mv_data;
         const time_t last_seen = averages->last_seen;
 
-        JsonElement *all_observables = JsonObjectCreate(observables_max);
-        assert(observables_max <= CF_OBSERVABLES);
+        GetObservableNames(tskey_filename, &obnames);
+        JsonElement *all_observables = JsonObjectCreate(CF_OBSERVABLES);
```

Comment:
`assert(obnames != NULL)` or run-time check with error? It probably "can't" happen since if no ts_key file then built-in should always work.